src/plasma/lib/python/plasma.py

```diff
@@ -258,30 +190,12 @@ def wait(self, object_ids, timeout, num_returns):
       ready_ids, waiting_ids (List[str], List[str]): List of object IDs that
         are ready and list of object IDs we might still wait on respectively.
     """
-    if not self.has_manager_conn:
-      raise Exception("Not connected to the plasma manager socket")
-    if num_returns < 0:
-      raise Exception("The argument num_returns cannot be less than one.")
-    if num_returns > len(object_ids):
-      raise Exception("The argument num_returns cannot be greater than len(object_ids): num_returns is {}, len(object_ids) is {}.".format(num_returns, len(object_ids)))
-    if timeout > 2 ** 36:
-      raise Exception("The method wait currently cannot be used with a timeout greater than 2 ** 36.")
-    object_id_array = (len(object_ids) * PlasmaID)()
-    for i, object_id in enumerate(object_ids):
-      object_id_array[i] = make_plasma_id(object_id)
-    return_id_array = (num_returns * PlasmaID)()
-    num_return_objects = self.client.plasma_wait(self.plasma_conn,
-                                                 object_id_array._length_,
-                                                 object_id_array,
-                                                 ctypes.c_int64(timeout),
-                                                 num_returns,
-                                                 return_id_array)
-    ready_ids = map(plasma_id_to_str, return_id_array[num_returns-num_return_objects:])
-    return ready_ids, list(set(object_ids) - set(ready_ids))
+    ready_ids, waiting_ids = libplasma.wait(self.conn, object_ids, timeout, num_returns)
+    return ready_ids, list(waiting_ids)
```

Comment:
I'd prefer to cast the set to a list in the C extension, but I didn't see an obvious method for this.